XenAdmin/Network/XenConnectionUI.cs

```diff
@@ -79,61 +79,42 @@ public static void BeginConnect(IXenConnection connection, bool interactive, For
                 ((XenConnection)connection).BeginConnect(initiateMasterSearch, PromptForNewPassword);
         }
 
-        public static bool PromptForNewPassword(IXenConnection connection, string oldPassword)
+        private static bool PromptForNewPassword(IXenConnection connection, string oldPassword)
         {
             bool result = false;
-            Program.Invoke(Program.MainWindow, delegate()
-                                                   {
-                                                       // show an altered version of the add server dialog with the hostname greyed out
-                                                       AddServerDialog dialog = new AddServerDialog(connection, true);
-                                                       result = dialog.ShowDialog(Program.MainWindow) == DialogResult.OK;
-                                                   });
+            Program.Invoke(Program.MainWindow, () =>
+            {
+                // show an altered version of the add server dialog with the hostname greyed out
+                using (var dialog = new AddServerDialog(connection, true))
+                    result = dialog.ShowDialog(Program.MainWindow) == DialogResult.OK;
+            });
             return result;
         }
 
         private static void RegisterEventHandlers(IXenConnection connection)
         {
             UnregisterEventHandlers(connection);
-            connection.ConnectionResult += Connection_ConnectionResult;

```

Comment:
With this change the connection result is not handled when the connection is attempted in the non-interactive mode, for example from ConnectAllHostsCommand

Reply:
To recap: the 3rd commit was reworked. A couple of additional commits were pushed to fix a typo and simplify event signatures.